scripts/provisioner.sh

```diff
@@ -142,11 +142,24 @@ fi
 if [[ "${DISABLE_KUBE_PROXY}" == "true" ]]; then
   kubectl -n "${KUBE_PROXY_NAMESPACE}" delete daemonset kube-proxy || true
   kubectl -n "${KUBE_PROXY_NAMESPACE}" delete cm kube-proxy || true
-  kubectl wait --timeout=300s --for=condition=Ready pod -l k8s-app=cilium -n "${CILIUM_NAMESPACE}" 
+  max_retry=60
+  for ((i=1; i<=max_retry; i++)); do
+    if ((i==max_retry)); then
+      echo "❌ Timeout: Cilium pod was not created after timeout."
+      exit 1
+    fi
+    if kubectl get pod -n "${CILIUM_NAMESPACE}" -l k8s-app=cilium 2>/dev/null | grep -q cilium; then
```

Comment:
why not use wait on the ds rollout? 



Simpler method from https://starkandwayne.com/blog/silly-kubectl-trick-5-waiting-for-things-to-finish-up-2/ ->

kubectl rollout status daemonset \
  cilium \
  -n kube-system \
  --timeout 300s

Reply:
I have thought this, if we wait for ds rollout, we will need more logic to check if the ds is available

Reply:
if the ds isnt available the pods arent available?

Reply:
yes, that's why we have the retry logic here.



Maybe let me explain here a bit.

we do `kubectl apply -f cilium-ds.yaml -n kube-system`

I would like to wait to have the cilium pod(so I dont have to care about DS status or if it's available  the API server)

Then I will wait for cilium pod is ready.